instrumentation/sofa-rpc-5.4/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/sofarpc/v5_4/TracingFilter.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.sofarpc.v5_4;
+
+import static com.alipay.sofa.rpc.common.RpcConstants.HIDE_KEY_PREFIX;
+
+import com.alipay.sofa.rpc.common.RemotingConstants;
+import com.alipay.sofa.rpc.config.AbstractInterfaceConfig;
+import com.alipay.sofa.rpc.config.ConsumerConfig;
+import com.alipay.sofa.rpc.context.RpcInternalContext;
+import com.alipay.sofa.rpc.core.exception.SofaRpcException;
+import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.alipay.sofa.rpc.filter.Filter;
+import com.alipay.sofa.rpc.filter.FilterInvoker;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+
+final class TracingFilter extends Filter {
+
+  private final Instrumenter<SofaRpcRequest, SofaResponse> instrumenter;
+  private final boolean isClientSide;
+
+  /** Hidden key: .otel.async.context otel Asynchronous call context */
+  private static final String HIDDEN_KEY_ASYNC_CONTEXT = HIDE_KEY_PREFIX + "otel_async_context";
+
+  TracingFilter(Instrumenter<SofaRpcRequest, SofaResponse> instrumenter, boolean isClientSide) {
+    this.instrumenter = instrumenter;
+    this.isClientSide = isClientSide;
+  }
+
+  @Override
+  @SuppressWarnings("ThrowsUncheckedException")
+  public SofaResponse invoke(FilterInvoker invoker, SofaRequest request) throws SofaRpcException {
+    if (shouldSkipLocalCall(invoker)) {
+      return invoker.invoke(request);
+    }
+
+    Context parentContext = Context.current();
+    SofaRpcRequest sofaRpcRequest = SofaRpcRequest.create(request);
+
+    if (!instrumenter.shouldStart(parentContext, sofaRpcRequest)) {
+      return invoker.invoke(request);
+    }
+    Context context = instrumenter.start(parentContext, sofaRpcRequest);
+
+    SofaResponse response;
+    boolean isSynchronous = true;
+    try (Scope ignored = context.makeCurrent()) {
+      response = invoker.invoke(request);
+      if (isClientSide && request.isAsync()) {
+        isSynchronous = false;
+        RpcInternalContext internalCtx = RpcInternalContext.getContext();
+        internalCtx.setAttachment(HIDDEN_KEY_ASYNC_CONTEXT, context);
```

Comment:
We usually use `VirtualField` of OTel to store context in async scenarios, you can take a look related case like:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/8afb6894146a7dfe8e3986434904d0a3d01ca52c/instrumentation/async-http-client/async-http-client-1-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/asynchttpclient/common/ResponseInstrumentation.java#L55-L59